airbyte-integrations/connectors/source-freshdesk/source_freshdesk/api.py

```diff
@@ -138,15 +138,58 @@ def _api_get(self, url: str, params: Mapping = None):
     def list(self, fields: Sequence[str] = None) -> Iterator[dict]:
         """Iterate over entities"""
 
+    @property
+    def name(self):
+        """Name of the stream"""
+        stream_name = self.__class__.__name__
+        if stream_name.endswith("API"):
+            stream_name = stream_name[:-3]
+        return stream_name
+
     def read(self, getter: Callable, params: Mapping[str, Any] = None) -> Iterator:
         """Read using getter"""
         params = params or {}
-        for page in range(1, self.maximum_page):
-            batch = list(getter(params={**params, "per_page": self.result_return_limit, "page": page}))
-            yield from batch
 
-            if len(batch) < self.result_return_limit:
-                break
+        stream = self.name
+        # This block extends TicketsAPI Stream to overcome '300 page' server error.
+
+        if stream == "Tickets":
+            page_break = 300  # the maximum page allowed to pull during pagination.
+
+            for page in range(1, self.maximum_page):
+                if page < page_break:
+                    batch = list(getter(params={**params, "per_page": self.result_return_limit, "page": page}))
+                    yield from batch
+                    last_record = batch[0]["updated_at"]  # position 0, because the records are returned in 'desc'
+
+                    if len(batch) < self.result_return_limit:
```

Comment:
Yes, this is not my implementation, so I had to figure out how that implementation and pagination works as well, basically, it says: 
`whatever you have fewer results from the last batch than expected return limit break the loop since we already fetched them, and nothing to return`
(all that according to the console output I've seen during this bugfix)

Reply:
makes sense; thanks